src/main/java/org/sonarqube/gradle/AndroidUtils.java

```diff
@@ -62,13 +65,22 @@ static void configureForAndroid(Project project, String variantName, final Map<S
     BaseVariant variant = findVariant(project, variantName);
     if (variant != null) {
       List<File> bootClassPath = getBootClasspath(project);
-      populateSonarQubeProps(properties, bootClassPath, variant, false);
-      if (variant instanceof TestVariant) {
+      if (project.getPlugins().hasPlugin("com.android.test")) {
+        // Instrumentation tests only
         populateSonarQubeProps(properties, bootClassPath, variant, true);
-      } else if (variant instanceof TestedVariant) {
-        UnitTestVariant unitTestVariant = ((TestedVariant) variant).getUnitTestVariant();
-        if (unitTestVariant != null) {
-          populateSonarQubeProps(properties, bootClassPath, unitTestVariant, true);
+      } else {
+        populateSonarQubeProps(properties, bootClassPath, variant, false);
+        if (variant instanceof TestedVariant) {
+          // Local tests
+          UnitTestVariant unitTestVariant = ((TestedVariant) variant).getUnitTestVariant();
+          if (unitTestVariant != null) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)